hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java

```diff
@@ -222,16 +228,17 @@ private static void waitForLogIndex(long minOMDBFlushIndex,
    * This means that an OM cannot support 2 prepare requests in the
    * transaction pipeline (un-applied) at the same time.
    */
-  public static void takeSnapshotAndPurgeLogs(long prepareIndex,
+  public static void takeSnapshotAndPurgeLogs(long lastLogIndex,
       RaftServer.Division division) throws IOException {
     StateMachine stateMachine = division.getStateMachine();
     long snapshotIndex = stateMachine.takeSnapshot();
+    LOG.info("takeSnapshot at {} for lastLogIndex {}", snapshotIndex, lastLogIndex);
 
-    if (snapshotIndex < prepareIndex) {
+    if (snapshotIndex != lastLogIndex) {
```

Comment:
I think changing the `<` to `!=` will cause prepare failures if a leader change happens after the snapshot is taken. See the description of #2288

Reply:
@errose28 , you are right.  The log index may change so  `takeSnapshot()` could return a higher index.  Good catch!